NitroxClient/Communication/Packets/Processors/InitialPlayerSyncProcessor.cs

```diff
@@ -438,6 +428,62 @@ public void CreateRemotePlayers(object sender, EventArgs eventArgs)
             }
         }
 
+        private void SpawnVehiclesAfterBasePiecesFinish(List<VehicleModel> vehicleModels, bool basePiecesToSpawn)
+        {
+            Log.Info("Received initial sync packet with " + vehicleModels.Count + " vehicles.");
+
+            if (vehicleModels.Count == 0)
+            {
+                return;
+            }
+
+            VehicleCreator vehicleCreator = new VehicleCreator(vehicleModels, vehicles);
+
+            if(basePiecesToSpawn)
+            {
+                ThrottledBuilder.main.QueueDrained += vehicleCreator.CreateVehicles;
+            }
+            else
+            {
+                vehicleCreator.CreateVehicles(null, null);
+            }
+        }
+
+        private class VehicleCreator
+        {
+            private List<VehicleModel> vehicleModels;
+            private Vehicles vehicles;
+
+            public VehicleCreator(List<VehicleModel> vehicleModels, Vehicles vehicles)
+            {
+                this.vehicleModels = vehicleModels;
+                this.vehicles = vehicles;
+            }
+
+            public void CreateVehicles(object sender, EventArgs eventArgs)
+            {
+                ThrottledBuilder.main.QueueDrained -= CreateVehicles;
+
+                List<VehicleModel> cyclopses = vehicleModels
+                    .Where(x => x.TechType == TechType.Cyclops)
+                    .ToList();
+
+                List<VehicleModel> nonCyclopses = vehicleModels
+                    .Where(x => !cyclopses.Any(x2 => x2.TechType == x2.TechType))
```

Comment:
I think there's a bug here: `x2.TechType == x2.TechType` should be `x.TechType == x2.TechType`

Reply:
Oops, yeah you are right. Will fix.